test/tools/d_do_test.d

```diff
@@ -99,6 +99,13 @@ struct EnvData
     bool dobjc;
     bool coverage_build;
     bool autoUpdate;
+    bool hasMicrosoftStudioCompiler() const
+    {
+        import std.uni : sicmp;
+        import std.range : only;
+        alias lowerCasePred = (a, b) => a.only.sicmp(b.only) == 0;
+        return ccompiler.endsWith!lowerCasePred("cl.exe");
+    }
```

Comment:
Wasn't sure about this. I think there are pros (lazy, encapsulated) and cons (only set once) to it.
I'm happy to drop this bit if it's controversial.

Reply:
This variable is checked many times (especially when there are alot of test combinations) and is ALWAYS checked at least once, so lazy doesn't really make sense.

I'd just make this a field like everything else and assign it below.

Reply:
Also, maybe change the field to `usingMicrosoftCompiler`

Reply:
Wait...I misread the code...looks like lazy might make sense...

Reply:
From https://forum.dlang.org/post/p6oibo$1lmi$1@digitalmars.com

> 19. Try to use function prefixes:
> 'is' is the parameter in a certain category?
> 'has' does the parameter have a certain feature?
> 'can' for can I do X with the parameter?

There are multiple features (aka compilers: D, C++, Objective C, ...), so "has" sounded to me like the best name to query for the feature's existence.

Reply:
But I guess the method parameter is really the "environment", not the compiler itself....  I must say that "has" is a bit confusing...

The least confusing might be `usingMicrosoftCompiler`...saying that  the environment is "using" the Microsoft compiler.

Reply:
Yes, but you query `envData` which has multiple compilers (`compiler`, `ccompiler`, and `dobjc`), thus my reasoning, but I don't feel strongly about it. I think both are a lot better than `msc` ;-)

Reply:
Ok changed.

Reply:
Not sure about the laziness though...I'll let you decide but it makes me a bit uneasy :)